### DISCUSSION TITLE
Does V currently support callback from C?

### DISCUSSION THREAD

**Discussion by susugagalala** [vlang/v#21374] [Questions and Answers]
I want to develop event-driven programs in V which calls a C toolkit which interacts with the user generating callbacks. The program captures these events by having those callbacks invoke user-written V functions. Is this currently possible with V?

Go supports this feature with cgo via the //export facility. If V has a similar facility or otherwise supports this callback mechanism, some code examples will be appreciated.

**Comment by susugagalala** [comment]
The sample call_v_from_c directory from examples folder shows creating a shared library from a V source file containing export directives (similar to //export in Go). Can a static library or archive be created instead of a shared library? I tried passing -static in lieu of -shared but got an error.

**Comment by JalonSolov** [comment]
https://github.com/vlang/v/blob/master/doc/docs.md#v-and-c the first example is calling sqlite from V, and providing a V callback (named `my_callback` in the example).
